framework/src/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaModelConversion.scala

```diff
@@ -69,14 +74,15 @@ private[server] class AkkaModelConversion(
         headers),
       request.method.name,
       new RequestTarget {
-        override lazy val uri: URI = new URI(uriString)
-        override lazy val uriString: String = _uriString
+        override lazy val uri: URI = new URI(headers.uri)
+        override lazy val uriString: String = headers.uri
```

Comment:
I think I mentioned elsewhere that this could be a `def` if `headers.uri` is cheap to call.